src/App.tsx

```diff
@@ -1,24 +1,304 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  postTodo,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+import { FilterType } from './types/FilterType';
 
-const USER_ID = 0;
+const USER_ID = 10824;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [errorType, setErrorType] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [indexUpdatedTodo, setIndexUpdatedTodo] = useState(0);
+  const [editedTodo, setEditedTodo] = useState<Todo | null>(null);
+
+  const fetchData = async () => {
+    try {
+      const newTodos = await getTodos(USER_ID);
+
+      setTodos(newTodos);
+    } catch (error) {
+      setErrorType('Unable to load todos');
+    }
+  };
+
+  useEffect(() => {
+    setTimeout(() => {
+      setErrorType('');
+    }, 3000);
+  }, [errorType]);
+
+  const onChangeQuery = (event: React.FormEvent<HTMLInputElement>) => {
+    setQuery(event.currentTarget.value);
+  };
+
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+
+  const getVisibleTodo = (filter: FilterType, todosArr: Todo[]) => {
+    switch (filter) {
+      case FilterType.ACTIVE:
+        return todosArr.filter(todo => !todo.completed);
+      case FilterType.COMPLETED:
+        return todosArr.filter(todo => todo.completed);
+      default:
+        return todosArr;
+    }
+  };
+
+  const deleteErrorMessage = () => {
+    setErrorType('');
+  };
+
+  const visibleTodos = getVisibleTodo(filterType, todos);
+
+  const hasCompleted = visibleTodos.some(todo => todo.completed);
+
+  (function handleTempTodo() {
+    if (tempTodo) {
+      visibleTodos.splice(indexUpdatedTodo, 1, tempTodo);
+    }
+  }());
```

Comment:
looks like redundant function. splice mutate visibleTodos array.
anyway it is better to make in useEffect.
just render tempTodo after visible todos
```suggestion
```
it can be a reason of this bug (blinking)
![chrome-capture-2023-5-26 (1)](https://github.com/mate-academy/react_todo-app-with-api/assets/33029864/64f35ff9-a85f-4ba8-b44f-d5a404cdc72f)